### PR TITLE
Implement new BBE for tcp secureSocket

### PR DESCRIPTION
**Pull request by MohamedSabthar**
$title

### CODE REVIEW

File: examples/tcp-transport-security/tcp_listener.bal [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import ballerina/tcp;
+import ballerina/io;
+
+// An tcp listener can be configured to communicate through SSL/TLS as well.
+// [secureSocket](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/tcp/latest/tcp/records/ListenerSecureSocket) record provides the SSL related configurations.
+// which will configure a listener to accept new connections that
+// are secured via SSL.
+service on new tcp:Listener(9090, secureSocket = {
+    // Provide the server certificate path and the privateKey path 
+    // keystore path along with keystore password.
```
Comment (praneesha) [inline examples/tcp-transport-security/tcp_listener.bal:10]:
```suggestion
    // of the keystore path along with the keystore password.
```

File: examples/tcp-transport-security/tcp_listener.out [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# To start the service and run the main function, navigate to the directory that contains the
+# `.bal` file and use the `bal run` command  below along with the Ballerina home path as a config.
```
Comment (praneesha) [inline examples/tcp-transport-security/tcp_listener.out:2]:
```suggestion
# `.bal` file and execute the `bal run` command below along with the Ballerina home path as a config.
```

File: examples/tcp-transport-security/tcp_client.out [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# To start the service and run the main function, navigate to the directory that contains the
+# `.bal` file and use the `bal run` command  below along with the Ballerina home path as a config.
```
Comment (praneesha) [inline examples/tcp-transport-security/tcp_client.out:2]:
```suggestion
# `.bal` file and execute the `bal run` command below along with the Ballerina home path as a config.
```

File: examples/tcp-transport-security/tcp-transport-security.description [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// This example demonstrates how the Ballerina tcp client can be configured to
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp-transport-security.description]: udpated

File: examples/tcp-transport-security/tcp-transport-security.description [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+// connect to an SSL/TLS listener through 1-way SSL/TLS connection
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp-transport-security.description]: updated

File: examples/tcp-transport-security/tcp-transport-security.description [hunk 6]
```diff
@@ -0,0 +1,7 @@
+// This example demonstrates how the Ballerina tcp client can be configured to
+// connect to an SSL/TLS listener through 1-way SSL/TLS connection 
+// (i.e., the server is verified by the client). This example uses the Ballerina
+// tcp listener to host a service and the tcp client sends
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp-transport-security.description]: updated

File: examples/tcp-transport-security/tcp-transport-security.description [hunk 7]
```diff
@@ -0,0 +1,7 @@
+// This example demonstrates how the Ballerina tcp client can be configured to
+// connect to an SSL/TLS listener through 1-way SSL/TLS connection 
+// (i.e., the server is verified by the client). This example uses the Ballerina
+// tcp listener to host a service and the tcp client sends 
+// requests for that listener.
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp-transport-security.description]: updated

File: examples/tcp-transport-security/tcp-transport-security.metatags [hunk 8]
```diff
@@ -0,0 +1,2 @@
+description: BBE on how the Ballerina TCP client and TCP listener listener can be configured to use 1-way SSL connection (i.e. the server is verified by the client)
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp-transport-security.metatags]: updated

File: examples/tcp-transport-security/tcp_client.bal [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import ballerina/tcp;
+import ballerina/io;
+
+public function main() returns error? {
+    // [secureSocket](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/tcp/latest/tcp/records/ClientSecureSocket) record used to configure the client with TLS
+    tcp:Client socketClient = check new ("localhost", 9002, secureSocket = {
+        // Provide the trusted certificate path or the truststore path 
+        // along with truststore password.
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp_client.bal]: updated

File: examples/tcp-transport-security/tcp_listener.bal [hunk 10]
```diff
@@ -0,0 +1,36 @@
+import ballerina/tcp;
+import ballerina/io;
+
+// An tcp listener can be configured to communicate through SSL/TLS as well.
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp_listener.bal]: updated

File: examples/tcp-transport-security/tcp_listener.bal [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+// [secureSocket](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/tcp/latest/tcp/records/ListenerSecureSocket) record provides the SSL related configurations.
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp_listener.bal]: updated

File: examples/tcp-transport-security/tcp_client.bal [hunk 12]
```diff
@@ -0,0 +1,25 @@
+import ballerina/tcp;
+import ballerina/io;
+
+public function main() returns error? {
+    // [secureSocket](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/tcp/latest/tcp/records/ClientSecureSocket) record used to configure the client with TLS
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp_client.bal]: updated

File: examples/tcp-transport-security/tcp_listener.bal [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+// which will configure a listener to accept new connections that
+// are secured via SSL.
+service on new tcp:Listener(9090, secureSocket = {
+    // Provide the server certificate path and the privateKey path
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp_listener.bal]: updated

File: examples/tcp-transport-security/tcp_listener.bal (hunk 1 above)
Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp_listener.bal]: updated

File: examples/tcp-transport-security/tcp_listener.out (hunk 2 above)
Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp_listener.out]: updated

File: examples/tcp-transport-security/tcp_client.out (hunk 3 above)
Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp_client.out]: updated

**Review by praneesha — APPROVED** [review]

File: examples/tcp-transport-security/tcp_listener.bal [hunk 14]
```diff
@@ -0,0 +1,39 @@
+import ballerina/tcp;
+import ballerina/io;
+
+// A TCP listener can be configured to communicate through SSL/TLS as well.
+// The [secureSocket](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/tcp/latest/tcp/records/ListenerSecureSocket) record provides the SSL related configurations,
+// which will configure a listener to accept new connections that
+// are secured via SSL.
+tcp:ListenerSecureSocket secureSocket = {
```
Comment (shafreenAnfar) [inline examples/tcp-transport-security/tcp_listener.bal:8]: listenerSecureSocket

Comment (MohamedSabthar) [reply examples/tcp-transport-security/tcp_listener.bal:8]: updated

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**